common/src/main/java/io/netty/util/internal/NativeLibraryLoader.java

```
@@ -148,6 +149,21 @@ private static boolean isOSX() {
     }
 
     /**
+     * Loads the first available library in the collection with the specified
+     * {@link java.lang.ClassLoader}
+     */
+    public static void loadFirstAvailable(Collection<String> names, ClassLoader loader) {
+        for (String name : names) {
+            try {
+                NativeLibraryLoader.load(name, loader);
+                return;
+            } catch (Throwable t) {
+                logger.debug("Unable to load the library: " + name + '.', t);
+            }
+        }
```

maybe we could throw an exception after this for loop (no library loaded).